sli/tokenarray.h

```diff
@@ -29,7 +29,7 @@
 #include <typeinfo>
 #include <cstddef>
 #include <vector>
-#include <valarray>
+#include <algorithm>
```

Comment:
@heplesser : `TokenArray::rotate` uses the `std::rotate` function from `algorithm`